src/com/android/deskclock/ringtone/RingtonePickerActivity.java

```diff
@@ -526,11 +528,28 @@ public void onItemClicked(ItemAdapter.ItemViewHolder<?> viewHolder, int id) {
         }
     }
 
+    // TODO docs
+    private final class AddCustomRingtoneResult {
+
+        private final Uri mPlaybackUri;
+
+        private final String mTitle;
+
+        AddCustomRingtoneResult(Uri playbackUri, String title) {
+            mPlaybackUri = playbackUri;
+            mTitle = title;
+        }
+
+        Uri getPlaybackUri() { return mPlaybackUri; }
+        String getTitle() { return mTitle; }
+    }
+
     /**
+     * TODO change docs
      * This task locates a displayable string in the background that is fit for use as the title of
      * the audio content. It adds a custom ringtone using the uri and title on the main thread.
      */
-    private final class AddCustomRingtoneTask extends AsyncTask<Void, Void, String> {
+    private final class AddCustomRingtoneTask extends AsyncTask<Void, Void, AddCustomRingtoneResult> {
```

Comment:
question: Unless my understanding is completely wrong, the `mUri` here is conceptually `mOriginalUri`? The one that user selected?

It might benefit from a rename if that's the case.